File: src/main/java/destiny/data/gauquelin/Sport450Reader.java
/**
 * @author smallufo 
 * Created on 2009/3/8 at 上午 12:45:32
 */ 
package destiny.data.gauquelin;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.Serializable;
import java.net.URL;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Sport450Reader implements TextDataReader , Serializable
{
  protected List<GPerson> persons = Collections.synchronizedList(new ArrayList<GPerson>());
  
  public Sport450Reader()
  {
    URL url = getClass().getResource("sport450.txt");
    
    File file;
    BufferedReader bReader = null;
    
    try
    {
      file = new File(url.toURI());
      FileReader fReader = new FileReader(file);
      bReader = new BufferedReader(fReader);

      String line = null;
      while ((line = bReader.readLine()) != null)
      {
        if (!line.equals("") && line.charAt(0) != '#')
        {
          GPerson gp = ParseUtils.parseType3(line);
          gp.setCategory("sport");
          this.persons.add(gp);
        }
      }
    }
    catch (Exception e)
    {
      e.printStackTrace();
    }
    finally
    {
      try
      {
        bReader.close();
      }
      catch (IOException e)
      {
        e.printStackTrace();
      }
    }
  }
  
  @Override
  public List<GPerson> getPersons()
  {
    return persons;
  }
  
  public static void main(String[] args)
  {
    TextDataReader r = new Sport450Reader();
    int i=0;
    for(GPerson person : r.getPersons())
    {
      i++;
      System.out.println("["+i+"] : " +person);
    }
  }
}

File: src/main/java/destiny/data/gauquelin/Sport450Reader.kt
/**
 * @author smallufo
 * Created on 2009/3/8 at 上午 12:45:32
 */
package destiny.data.gauquelin

import destiny.data.gauquelin.ParseUtils.Companion.parseType3
import java.io.*

class Sport450Reader : TextDataReader, Serializable {


  override val persons: List<GPerson>
    get() {
      val ps = mutableListOf<GPerson>()
      val url = javaClass.getResource("sport450.txt")
      val file: File
      var bReader: BufferedReader? = null
      try {
        file = File(url.toURI())
        val fReader = FileReader(file)
        bReader = BufferedReader(fReader)
        var line: String?
        while (bReader.readLine().also { line = it } != null) {
          if (line != "" && line!![0] != '#') {
            val gp = parseType3(line!!)
            gp.category = "sport"
            ps.add(gp)
          }
        }
      } catch (e: Exception) {
        e.printStackTrace()
      } finally {
        try {
          bReader!!.close()
        } catch (e: IOException) {
          e.printStackTrace()
        }
      }
      return ps.toList()
    }

  companion object {
    @JvmStatic
    fun main(args: Array<String>) {
      val r = Sport450Reader()
      var i = 0
      for (person in r.persons) {
        i++
        println("[$i] : $person")
      }
    }
  }


}
